Add guidance for iOS campaign card component (#523)
nhsuk/nhsapp-frontend: docs/ios/campaign-card.md

---
layout: layouts/ios.njk
title: Campaign card
tags:
  - iosComponents
---

The campaign card is a tappable card that helps users learn about a public health campaign.

<img src="/assets/images/ios/campaign-card.png" width="375">

## When to use

Use a campaign card to promote prioritised public health campaigns that:

- are relevant to most NHS App users
- have an onward digital journey that meets the [standards for NHS App integration](https://digital.nhs.uk/services/nhs-app/how-to-integrate-with-the-nhs-app/standards-for-nhs-app-integration)

> Campaigns that are not run by the NHS should ideally have a journey that has been integrated with the NHS App.

Reviewer: Bit unclear what this means.

Reviewer 2: I think we could simplify this part to something like:

> Use a campaign card to promote prioritised public health campaigns. We promote campaigns that:
> 
> - are widely relevant to the broad range of people using the NHS App 
> - meet our [integration standards](https://digital.nhs.uk/services/nhs-app/how-to-integrate-with-the-nhs-app/standards-for-nhs-app-integration) if they take users to an external campaign website

That was the original spirit, from memory of the conversations with the product manager who set the criteria for what kinds of campaigns we'd actually feature on the campaign card. I could run this past them to double check, if you think this makes it a bit clearer?

Use one campaign card per screen at most.

## When not to use

Do not use a campaign card:

- to prompt the user to complete a task, such as giving feedback — use the [banner](/ios/banner)
- for navigation to services or the user's own records
- for information that does not have an onward journey

## How it works

A campaign card has:

- a full-width photograph
- a heading in bold

> - body text

Reviewer: Match the terminology from the component?

```suggestion
- a title in bold
- a subtitle, not in bold
```

The whole card is one tap target. The text sits on a dark blue panel. Text never overlaps the photograph.

The layout adapts to the user's device and settings:

- on narrow screens, the photograph sits above the text
- on wider screens, such as iPad, the photograph sits to the left of the text
- text scales with Dynamic Type
- at large text sizes, the photograph always sits above the text, so the text has the full width of the card
- text is never truncated: the panel grows to fit

<img src="/assets/images/ios/campaign-card-ipad.png">

## How to use

Use the campaign card within a SwiftUI view.

The card triggers an action when tapped. Specify this using the `action` closure. The action can navigate to another screen, present a sheet, open a link, or trigger any other action.

{% from "details/macro.njk" import details %}
{% call details({ summaryText: "Swift options" }) %}
{% include "ios/campaign-card/swift-options.md" %}
{% endcall %}

```swift { .nhsuk-code--button }
CampaignCard(
    image: Image("campaign-organ-donation"),
    title: "Organ donors save lives",
    subtitle: "Take 2 minutes to confirm your organ donation decision",

>     action: { path.append(.organDonation) }

Reviewer: I'm not sure how complete these code examples should be (a whole view, or just the component) - we’ve done a mixture so far.

As this one just has the component, could swap the the action out with a comment, or leave it empty?

```suggestion
    action: {
        // open a new view
    }
```

Alternative could add a complete view with all the `NavigationPath` setup, like https://design-system.nhsapp.service.nhs.uk/ios/home-menu/ does - but I'm starting to think that that’s a bit much for a component page?

Author: How about instead of leaving it empty we add something like...

```
action: { 
    // open the campaign journey
}
```

?

Author: It does feel like adding `NavigationPath` for each component example is a bit much. That's something that should be covered in examples on page (screen) layout guidance?

)
```

### Using a campaign card in a list view

To place a campaign card inside a `List`, apply the `nhsCardRowStyle()` modifier so it renders edge to edge rather than as a standard inset row:

```swift { .nhsuk-code--button }
List {
    CampaignCard(
        image: Image("campaign-organ-donation"),
        title: "Organ donors save lives",
        subtitle: "Take 2 minutes to confirm your organ donation decision",
        action: { path.append(.organDonation) }
    )
    .nhsCardRowStyle()
}
```

## Writing for this component

The heading should have a persuasive tone. The wording of existing headings or slogans for the campaign can be a helpful reference point. Aim for a maximum of 23 characters including spaces, so the heading fits on one line at standard text sizes.

> The body text should have an informative tone. It should help users understand what they can do if they follow the link. Aim for a maximum of 75 characters including spaces, so it fits on two lines at standard text sizes.

Reviewer: Update these to use the terminology from the component?

And are these intended be single sentences/phrases and not end in a full stop?

```suggestion
The title should have a persuasive tone. The wording of existing headings or slogans for the campaign can be a helpful reference point. Aim for a maximum of 23 characters including spaces, so the heading fits on one line at standard text sizes. Do not add a full stop.

The subtitle should have an informative tone. It should help users understand what they can do if they follow the link. Aim for a maximum of 75 characters including spaces, so it fits on two lines at standard text sizes. Keep as a single sentence and do not add a full stop.
```

Reviewer 2: The suggested changes here look good to me. The pattern with the campaign card has been to use single sentences/phrases, and to exclude the full stop. The full stop question was agreed at a content crit and with interaction designers, so good to put it in writing!

These character counts are targets, not hard limits. Text is never cut off: at larger text sizes, or with longer text, the card grows. Always check the card at the largest text sizes.

The accessibility hint is optional. Only add one if the heading and body text do not already make the result of tapping clear. For example, if the card opens a web page, a hint like "Opens in a web browser" helps.

Follow the [NHS content guide](https://service-manual.nhs.uk/content) for style, voice and tone.

## Accessibility

This component supports Dynamic Type, Dark Mode and VoiceOver.

VoiceOver reads the card as a single button: the heading and body text together, followed by the accessibility hint if one is set.

> The photograph is usually decorative rather than informative, so it is hidden from VoiceOver. Check the [service manual guidance on alt text](https://service-manual.nhs.uk/accessibility/content) to make sure this is right for your image.

Reviewer: Do we need to support an optional label for the image in case it's not strictly decorative?

Author: I'm not sure tbh. @Graham-Pembrey what do you think? Do you know what the previous campaigns have been and if they have used alt text?

Reviewer 2: Here are all the recent campaigns. We treated them all as decorative images and didn't include alt text, so that does seem to be the pretty consistent pattern.

Possibly the optional label might be something to leave out until a need arises?

<img width="246" height="256" alt="image" src="https://github.com/user-attachments/assets/104941ae-1701-43fd-8d01-62297442c182" />

<img width="266" height="275" alt="image" src="https://github.com/user-attachments/assets/b6c0e96b-61c1-467c-a054-2522f0ef110b" />

<img width="304" height="317" alt="image" src="https://github.com/user-attachments/assets/360cf84b-583e-4ebb-9bf5-f58662c29b4c" />

<img width="245" height="244" alt="image" src="https://github.com/user-attachments/assets/c648c543-bdc2-4f0f-919f-12d805b57150" />

<img width="272" height="290" alt="image" src="https://github.com/user-attachments/assets/47606a20-7991-46cd-b8e6-fb315dacba04" />

Author: Thanks @Graham-Pembrey - I'm happy to leave it for now and revisit it later if required.

---
layout: layouts/ios.njk
title: Campaign card
tags:
  - iosComponents
---

The campaign card is a tappable card that helps users learn about a public health campaign.

<img src="/assets/images/ios/campaign-card.png" width="375">

## When to use

Use a campaign card to promote prioritised public health campaigns that:

- are relevant to most NHS App users

> - have an onward digital journey that meets the [standards for NHS App integration](https://digital.nhs.uk/services/nhs-app/how-to-integrate-with-the-nhs-app/standards-for-nhs-app-integration)

Reviewer 2: ```suggestion
- meet the [standards for NHS App integration](https://digital.nhs.uk/services/nhs-app/how-to-integrate-with-the-nhs-app/standards-for-nhs-app-integration) if they take users to an external campaign website
```

> Campaigns that are not run by the NHS should ideally have a journey that has been integrated with the NHS App.

Reviewer 2: ```suggestion

```

Use one campaign card per screen at most.

## When not to use

Do not use a campaign card:

- to prompt the user to complete a task, such as giving feedback — use the [banner](/ios/banner)

> - for navigation to services or the user's own records

Reviewer 2: Although this is generally true, the latest campaign card goes to the organ donation service in the app, which maybe blurs the line a bit. I wonder if it would be simpler to leave this line out?
